### PR TITLE
Copyediting RLMCollectionChange, RLMCollection, and constants

### PR DESCRIPTION
**Pull request by austinzheng**
@bdash @tgoyne @jpsim @mrackwitz 

Changes:
- Made some additional changes to RLMArray documentation
- Formatted, cleaned up, and copyedited docstrings
- Added backticks for type names

### CODE REVIEW

File: Realm/RLMConstants.h [hunk 1]
```diff
@@ -94,47 +125,49 @@ typedef NS_ENUM(NSInteger, RLMError) {
 #pragma mark - Notification Constants
 
 /**
- Posted by RLMRealm when the data in the realm has changed.
+ This notification is posted by a Realm when the data in that Realm has changed.
 
- DidChange are posted after a realm has been refreshed to reflect a write
```
Comment (austinzheng) [inline Realm/RLMConstants.h]: Note that the old description was kind of screwed up.

File: Realm/RLMArray.h [hunk 2]
```diff
@@ -264,7 +263,7 @@ NS_ASSUME_NONNULL_BEGIN
  Returns a sorted `RLMResults` from the array.
 
  @param property    The property name to sort by.
- @param ascending   The direction to sort by.
+ @param ascending   The direction to sort.
```
Comment (jpsim) [inline Realm/RLMArray.h]: I think it was previously correct and appropriate to end on the preposition. "To sort by."

File: Realm/RLMCollection.h [hunk 3]
```diff
@@ -23,7 +23,7 @@ NS_ASSUME_NONNULL_BEGIN
 @class RLMRealm, RLMResults, RLMObject, RLMSortDescriptor, RLMNotificationToken, RLMCollectionChange;
 
 /**
- A homogenous collection of `RLMObject`s like `RLMArray` or `RLMResults`.
+ A homogenous collection of `RLMObject` instances. Examples of conforming types include `RLMArray` and `RLMResults`.
```
Comment (jpsim) [inline Realm/RLMCollection.h]: `RLMArray`, `RLMResults` and `RLMLinkingObjects`.

File: Realm/RLMCollection.h [hunk 4]
```diff
@@ -223,40 +223,41 @@ NS_ASSUME_NONNULL_BEGIN
...
 - (RLMNotificationToken *)addNotificationBlock:(void (^)(id<RLMCollection> __nullable collection,
                                                          RLMCollectionChange *__nullable change,
                                                          NSError *__nullable error))block __attribute__((warn_unused_result));
 
 @end
+
 /**
- An RLMSortDescriptor stores a property name and a sort order for use with
- `sortedResultsUsingDescriptors:`. It is similar to NSSortDescriptor, but supports
+ A `RLMSortDescriptor` stores a property name and a sort order for use with
```
Comment (jpsim) [inline Realm/RLMCollection.h]: we use "an" for all other RLM-prefixed types

File: Realm/RLMCollection.h [hunk 5]
```diff
@@ -223,40 +223,41 @@ NS_ASSUME_NONNULL_BEGIN
...
- Whether this descriptor sorts in ascending or descending order.
+ Whether the descriptor sorts in ascending or descending order.
  */
 @property (nonatomic, readonly) BOOL ascending;
 
 #pragma mark - Methods
 
 /**
- Returns a new sort descriptor for the given property name and order.
+ Returns a new sort descriptor for the given property name and sort order.
```
Comment (jpsim) [inline Realm/RLMCollection.h]: "sort direction"?

File: Realm/RLMConstants.h [hunk 6]
```diff
@@ -21,71 +21,102 @@
...
     RLMErrorFail                  = 1,
-    /** Returned by RLMRealm for any I/O related exception scenarios when a realm is opened. */
+
+    /** Denotes a file I/O error that occurred when trying to open a Realm. */
     RLMErrorFileAccess            = 2,
-    /** Returned by RLMRealm if the user does not have permission to open or create
-        the specified file in the specified access mode when the realm is opened. */
+
+    /** 
+     Denotes a file permissions error that ocurred when trying to open a Realm.
```
Comment (jpsim) [inline Realm/RLMConstants.h]: singular "permission"?

**Comment by jpsim** [comment]
We should be consistent with the Realm Swift documentation by mirroring any updates made here there too.

**Comment by austinzheng** [comment]
I'll be sure when I update the Swift docstrings to maintain consistency.

File: Realm/RLMArray.h (hunk 2 above)
Comment (bdash) [inline Realm/RLMArray.h]: I don't think it's correct to say that you "sort by" a direction. I'd say that you sort by a property _in_ a direction.

File: Realm/RLMConstants.h [hunk 7]
```diff
@@ -94,47 +125,49 @@ typedef NS_ENUM(NSInteger, RLMError) {
...
- Posted by RLMRealm when the data in the realm has changed.
+ This notification is posted by a Realm when the data in that Realm has changed.
 
- DidChange are posted after a realm has been refreshed to reflect a write
- transaction, i.e. when an autorefresh occurs, `[RLMRealm refresh]` is
- called, after an implicit refresh from `[RLMRealm beginWriteTransaction]`,
- and after a local write transaction is committed.
+ More specifically, this notification is posted after a Realm has been refreshed to
+ reflect a write transaction. This can happen when an autorefresh occurs, when
+ `[RLMRealm refresh]` is called, after an implicit refresh from `[RLMRealm beginWriteTransaction]`,
```
Comment (bdash) [inline Realm/RLMConstants.h]: This should be `-[RLMRealm refresh]` and `-[RLMRealm beginWriteTransaction]` respectively (leading dash to donate instance method).

**Comment by bdash** [comment]
👍

**Pull request merged**